src/screens/referScreen/referScreen.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { Fragment } from 'react';
+import { useEffect, useState } from 'react';
+import { useIntl } from 'react-intl';
+import { FlatList, Text, View } from 'react-native';
+import { BasicHeader, Icon, MainButton, UserListItem } from '../../components';
+import get from 'lodash/get';
+// utils
+import { getReferralsList } from '../../providers/ecency/ecency';
+import { Referral } from '../../models';
+
+// styles
+import styles from './referScreenStyles';
+
+const ReferScreen = ({ navigation }) => {
+  const intl = useIntl();
+  const [referralsList, setReferralsList] = useState<Referral[]>([]);
+  useEffect(() => {
+    _getReferralsList();
+  }, []);
+
+  console.log('-----referralsList----- : ', referralsList);
+
+  const _getReferralsList = async () => {
+    const referralsListData = await getReferralsList('good-karma');
+    setReferralsList(referralsListData as any);
+  };
+  const _renderPointsEarned = () => {
+    return (
+      <View style={styles.pointsEarnedContainer}>
+        <Text style={styles.points}>1000</Text>
+        <Text style={styles.earendText}>Points Earned</Text>
+        <MainButton
+          // isLoading={isLoading}
+          // isDisable={isLoading}
+          style={styles.mainButton}
+          height={50}
+          onPress={() => console.log('pressed!')}
+        >
+          <View style={styles.mainButtonWrapper}>
+            <Text style={styles.unclaimedText}>Refer</Text>
+            <View style={styles.mainIconWrapper}>
+              <Icon name="add" iconType="MaterialIcons" color="#357ce6" size={23} />
+            </View>
+          </View>
+        </MainButton>
+      </View>
+    );
+  };
+
+  const _renderReferralListItem = ({ item, index }: { item: Referral; index: number }) => {
+    return (
+      <UserListItem
+        key={get(item, '_id')}
+        index={index}
+        username={item.referredUsername}
+        description={get(item, 'created')}
+        isHasRightItem
+        isClickable
+        isBlackRightColor
+        rightText={item.isRewarded ? 'Rewarded' : 'Not Rewarded'}
```

Comment:
instead of Rewarded Unrewarded text, may we can show blue double dollar icon and gray double dollar icon... @feruzm what do you think... as for show filling bar, apparently the api do not return enough information for that and making api calls for each user may not be a good approach. double dollar can have tooltip to show what it represent

Delegate HP button on right..

Check screenshot for reference, also added figma

<img width="321" alt="Screenshot 2022-02-17 at 9 56 00 PM" src="https://user-images.githubusercontent.com/6298342/154531477-11071abb-bacf-4200-966e-a01965ccbd12.png">

Reply:
I like this approach, delegate is good to make new people feel welcomed. We will show tutor tooltip to give that hint and other details could be explained that way as well.